relay-server/src/services/project_cache.rs

```diff
@@ -852,7 +851,7 @@ impl ProjectCacheBroker {
         // state or we do not need one.
         if project_state.is_some()
             && (sampling_state.is_some() || sampling_key.is_none())
-            && !self.buffer_guard.is_over_high_watermark()
+            && self.memory_stat_config.has_enough_memory()
```

Comment:
In this new version of the spooler, we are just checking a single memory level to validate whether we want to perform certain actions. This will change once the new spooler will be implemented.